exercises/chapter6/test/Main.purs

```diff
@@ -1,17 +1,190 @@
 module Test.Main where
 
 import Prelude
-
+import Data.Foldable (foldMap, foldl, foldr)
+import Data.Hashable (hash)
+import Data.List (List(..), (:))
 import Effect (Effect)
-import Effect.Class.Console (logShow)
-import Data.Hashable (hash, hashEqual)
+import Partial.Unsafe (unsafePartial)
+import Test.Solutions (Complex(..), Extended(..), Hour(..), Multiply(..), NonEmpty(..), OneMore(..), Self(..), act)
+import Test.Unit (suite, test)
+import Test.Unit.Assert as Assert
+import Test.Unit.Main (runTest)
 
 main :: Effect Unit
-main = do
-  logShow (hash 123)
-  logShow (hash true)
-  logShow (hash [1, 2, 3])
-  logShow (hash "testing")
-  logShow (hash 'a')
-  logShow ("foo" `hashEqual` "foo")
-  logShow ("foo" `hashEqual` "bar")
+main =
+  runTest do
+    suite "Exercise Group 1" do
+      suite "Exercise 1 - Show Shape" do
+        -- This should probably be added as an exercise to the end of chapter 5
+        -- with a note to revisit as first exercise in chapter 6
+        test "Show Shape todo"
+          $ Assert.equal "todo" "todo" -- (show myShape)
+    suite "Exercise Group 2" do
+      suite "Exercise 1 - Show and Eq for Complex" do
+        test "Show Complex"
+          $ Assert.equal "1.0+2.0i"
+          $ show
+          $ Complex { real: 1.0, imaginary: 2.0 }
+ {-  Move this block comment starting point to enable more tests
```

Comment:
Readers can drag this line down to enable more tests as they tackle each exercise.